pkg/inference/memory/estimator.go

```diff
@@ -39,10 +39,11 @@ func (m *memoryEstimator) GetRequiredMemoryForModel(ctx context.Context, model s
 	return m.defaultBackend.GetRequiredMemoryForModel(ctx, model, config)
 }
 
-func (m *memoryEstimator) HaveSufficientMemoryForModel(ctx context.Context, model string, config *inference.BackendConfiguration) (bool, error) {
+func (m *memoryEstimator) HaveSufficientMemoryForModel(ctx context.Context, model string, config *inference.BackendConfiguration) (bool, *inference.RequiredMemory, *inference.RequiredMemory, error) {
```

Comment:
Should we try to unify pointer vs. non-pointer return types?  `inference.RequiredMemory` seems it should always be passed / returned as a non-pointer type since it's small (as it is in `SystemMemoryInfo`'s methods).